### PR TITLE
[Not ready] Add support for Bitlocker evidence

### PR DESCRIPTION
**Pull request by taishi8117**
Tested locally with BitlockerGo image by running 
* turbiniactl -C bitlocker -e /evidence/incoming/test.dd -r [recovery key]
* turbiniactl -C bitlocker -e /evidence/incoming/test.dd -p [password]

It is inefficient because Turbinia decrypts the disk as preprocess for every task. I will fix this in the next commit.

### CODE REVIEW

File: turbinia/evidence.py [hunk 1]
```diff
@@ -242,11 +243,68 @@ def __init__(
...
+  """
+
+  def __init__(
+      self, recovery_key=None, password=None, encrypted_path=None,
+      unencrypted_path=None, *args, **kwargs):
+    """Initialization for Bitlocker disk evidence object"""
+    self.recovery_key = recovery_key
+    self.password = password
+    self.encrypted_path = encrypted_path
+    self.unencrypted_path = unencrypted_path
```
Comment (ericzinnikas) [inline turbinia/evidence.py]: I think we can re-use some of the attributes from `EncryptedDisk` here. `unencrypted_path` can just get passed through `super` and you could probably move `encrypted_path` there as well (or just re-use `local_path` from RawDisk)?

**Comment by ericzinnikas** [comment]
I didn't think about this earlier, but we probably want to try and use a FUSE mount instead of re-writing the entire unencrypted disk. It looks like libbde supports this (with `bdemount`), but I'm not sure it is exposed through the python bindings: https://github.com/libyal/libbde/wiki/Mounting

Using FUSE would remove the need for the worker to have a large amount of local storage (for the unencrypted disk) and also massively speed up the process (not writing the entire disk out again).

I'm not sure how difficult this would be to integrate, because it seems like it will require installing the `libbde` binary, unless you can figure out a way to do it in pure Python.

As a bit of a stopgap, you could just use plaso's built in Bitlocker handling and simply add in a `--credential` argument to the `PlasoTask`, which would at least solve the plaso use-case.

**Comment by taishi8117** [comment]
Yeah I agree, I'll modify the PR to use a FUSE mount. I'll look into ways to do it in pure Python, but it might require installing libbde binary as you said.

**Comment by aarontp** [comment]
Thanks for the PR!  Let me know if you want me to review as is, or if I should wait until you do some investigation into the FUSE side of things.

**Comment by taishi8117** [comment]
Yes, I am implementing a FUSE integration right now - until that's completed, there is no need for review. I will rebase this once #349 is accepted, then will let you know once it's ready.

**Comment by aarontp** [comment]
I'm going to close this PR for now, but we can re-open if/when there are updates.  Thanks!

**Pull request closed without merging**